From a87ebed70e1d50097bad3b118e513830bb33a3c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Wed, 16 Mar 2022 22:15:30 +0100
Subject: [PATCH] Do not adjust compiler flags
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Just guesswork:
Something ha changed elsewhere causing the if-condition pass as seen for broken
build:

CMake compiler flags set to »-Wall -Wextra -Werror... - long but missing includes

Fixes build errors as:
| <..>/wesnoth-1.16.2/src/color.hpp:17:10: fatal error: algorithm: No such file or directory
| <..>/1.16.2-r0/wesnoth-1.16.2/src/tstring.hpp:18:10: fatal error: memory: No such file or directory

Upstream-Status: Inappropriate Inappropriate [oe specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 CMakeLists.txt | 16 ++--------------
 1 file changed, 2 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aaf6923..04efa09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,20 +218,8 @@ if(NOT WIN32)
 		set(COMPILER_FLAGS "${COMPILER_FLAGS} -fno-omit-frame-pointer -fno-optimize-sibling-calls")
 	endif()
 
-### Set the final compiler flags.
 
-	set(COMPILER_FLAGS "${COMPILER_FLAGS} ${CXX_FLAGS_USER}")
-
-	if(NOT "${CMAKE_CXX_FLAGS}" STREQUAL "${COMPILER_FLAGS}")
-		MESSAGE(STATUS "CMake compiler flags set to »${COMPILER_FLAGS}«")
-		set(CMAKE_CXX_FLAGS
-			"${COMPILER_FLAGS}"
-			CACHE
-			STRING
-			"Global flags used by the CXX compiler during all builds."
-			FORCE
-		)
-	endif()
+# do not align compiler flag suggested by oe
 
 # #
 # Determine optimization level
@@ -416,7 +404,7 @@ if(NOT WIN32)
 # End setting profiler build options
 # #
 else()
-	set(CMAKE_CXX_FLAGS "/W3 /WX /wd4503 /wd4351 /wd4250 /wd4244 /wd4267 /we4239 /wd4275 /EHsc /utf-8" CACHE STRING "Global flags used by the CXX compiler during all builds." FORCE)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3 /WX /wd4503 /wd4351 /wd4250 /wd4244 /wd4267 /we4239 /wd4275 /EHsc /utf-8" CACHE STRING "Global flags used by the CXX compiler during all builds." FORCE)
 	add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WIN7 -D_CRT_SECURE_NO_WARNINGS -D_WINSOCK_DEPRECATED_NO_WARNINGS -DNOMINMAX)
 
 	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG_LUA")
-- 
2.34.1

